fhir/com.b2international.snowowl.fhir.rest/src/com/b2international/snowowl/fhir/rest/FhirConceptMapTranslateOperationController.java

```diff
@@ -0,0 +1,210 @@
+	}
+	
+	/**
+	 * HTTP POST request to translate a code that belongs to a {@link ConceptMap} specified by its ID.
+	 * @param conceptMapId
+	 * @return translation of the code
+	 */
+	@Operation(
+		summary="Translate a code based on a specific Concept Map",
+		description="Translate a code from one value set to another, based on the existing value set and specific concept map."
+	)
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+		@ApiResponse(responseCode = "404", description = "Not found"),
+		@ApiResponse(responseCode = "400", description = "Bad request")
+	})
+	@PostMapping(value="/{conceptMapId:**}/$translate", consumes = AbstractFhirResourceController.APPLICATION_FHIR_JSON)
+	public Promise<Parameters.Fhir> translate(
+		@Parameter(description = "The id of the conceptMap to base the translation on") 
+		@PathVariable("conceptMapId") 
+		String conceptMapId,
+		
+		@Parameter(description = "The translate request parameters")
+		@RequestBody 
+		Parameters.Fhir body) {
+		
+		//validation is triggered by builder.build()
+		final TranslateRequest request = toRequest(body, TranslateRequest.class);
+		request.setUrl(conceptMapId);
+		return doTranslate(request);
+	}
+
+	/**
+	 * HTTP Get request to translate a code that could belongs to any {@link ConceptMap} in the system.
+	 * @return translation of the code
+	 */
+	@Operation(
+		summary="Translate a code",
+		description="Translate a code from one value set to another, based on the existing value set and concept maps resources, and/or other additional knowledge available to the server."
+	)
+	@ApiResponses({
+		@ApiResponse(responseCode = "200", description = "OK"),
+		@ApiResponse(responseCode = "400", description = "Bad request"),
+		@ApiResponse(responseCode = "404", description = "Concept map not found")
+	})
+	@GetMapping(value="/$translate")
+	public Promise<Parameters.Fhir> translate(
+		@Parameter(description = "The code to translate") @RequestParam(value="code") final String code,
+		@Parameter(description = "The code system's uri") @RequestParam(value="system") final String system,
+		@Parameter(description = "The code system's version, if null latest is used") @RequestParam(value="version") final Optional<String> version,
+		@Parameter(description = "The source value set") @RequestParam(value="source") final Optional<String> source,
+		@Parameter(description = "Value set in which a translation is sought") @RequestParam(value="target") final Optional<String> target,
+		@Parameter(description = "Target code system") @RequestParam(value="targetsystem") final Optional<String> targetSystem,
+		@Parameter(description = "If true, the mapping is reversed") @RequestParam(value="reverse") final Optional<Boolean> isReverse) {
+		
+		//validation is triggered by builder.build()
+		Builder builder = TranslateRequest.builder()
+			.code(code)
+			.system(system);
+		
+		if (version.isPresent()) {
+			builder.version(version.get());
+		}
```

Comment:
`Optional` offers a shortcut for these blocks, it should also work with a method reference as the consumer:

```suggestion
		version.ifPresent(builder::version);
```

Reply:
Yeah, I have just copy-pasted the original code 😅, but will apply these improvements.